### PR TITLE
Ingest an article with a digest

### PR DESCRIPTION
**Pull request by giorgiosironi**
For https://github.com/elifesciences/issues/issues/4326

This test may "pass" at the moment, but only because we don't do any check after ingesting this article.

### CODE REVIEW

**Comment by giorgiosironi** [comment]
Had to clear the logs due to disk space problems.

File: spectrum/checks.py [hunk 1]
```diff
@@ -349,6 +349,9 @@ def annotations(self, profile_id, access='public'):
     def digests(self):
         return self._list_api('/digests', 'digest')
 
+    def digest(self, id):
+        return self._item_api('/digests/%s' % id, 'digest')
```
Comment (gnott) [inline spectrum/checks.py]: I'm not sure if a `GET` will work here if the content is `"stage": "preview"` in the digests API without providing an `Authorization` header. Temporarily, I think the `PUT` to the endpoint is not working just yet.

Comment (giorgiosironi) [reply spectrum/checks.py]: Can't see it from here, but the instance of `ApiCheck` I am using has the `Authorization` header passed in the constructor.

File: spectrum/test_digest.py [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
-def test_digest_lifecycle(generate_digest):
+def test_digest_lifecycle(generate_digest, generate_article):
     digest = generate_digest(SIMPLEST_ARTICLE_ID)
     input.DIGESTS_BUCKET.upload(digest.filename(), id=digest.article_id())
     checks.BOT_EMAILS.wait_email(subject='Digest: Anonymous_%s' % digest.article_id())
     checks.BOT_INTERNAL_DIGEST_OUTBOX_DOC.of(id=digest.article_id())
     checks.BOT_INTERNAL_DIGEST_OUTBOX_JPG.of(id=digest.article_id())
     #checks.DIGEST_JPG_PUBLISHED_CDN_BUCKET.of(id=digest.article_id())
+
+    article = generate_article(template_id=SIMPLEST_ARTICLE_ID, article_id=digest.article_id())
```
[the remark this answers is not in the document]
Comment (giorgiosironi) [reply spectrum/test_digest.py]: Checking what other one we could use, hopefully that doesn't have images to speed things up.

**Comment by giorgiosironi** [comment]
Last iteration:
```
2018-09-24T11:01:52Z ERROR worker_1518 Exception issuing PUT to the digest endpoint for digest_id 5425210691942615893. Details: Error put digest 5425210691942615893 to digest API: 400
{"title":"image.thumbnail = 'size' is a required property"}
Traceback (most recent call last):
  File "/opt/elife-bot/activity/activity_IngestDigestToEndpoint.py", line 280, in put_digest_to_endpoint
    digest_provider.put_digest(digest_id, digest_content, settings)
  File "/opt/elife-bot/provider/digest_provider.py", line 184, in put_digest
    digest_auth_key(settings, auth))
  File "/opt/elife-bot/provider/digest_provider.py", line 175, in digest_put_request
    (status_code, response.content))
ErrorCallingDigestException: Error put digest 5425210691942615893 to digest API: 400
{"title":"image.thumbnail = 'size' is a required property"}
```
Looks like a IIIF failure, will debug; and also change the template article to avoid issues with `body`.

**Comment by giorgiosironi** [comment]
`elife-bot--end2end` is misconfigured to talk to `iiif--prod`:
```
2018-09-24 12:48:51,586 ERROR Exception in iiif_server_info for GET https://iiif.elifesciences.org/digests/8253269141543206847%2Fdigest-8253269141543206847.jpg/info.json. Details: No JSON object could be decoded
```

**Comment by giorgiosironi** [comment]
This worked only by chance on `continuumtest` because the image is available in `prod`:
https://iiif.elifesciences.org/digests:33947/digest-33947.jpg/full/full/0/default.jpg

**Comment by giorgiosironi** [comment]
Good test, I think, in `end2end`:
```
2018-09-24T13:15:39Z INFO worker_4064 Put to digest API: PUT http://end2end--gateway.elife.internal/digests/9149295672428006847
OrderedDict([('id', '9149295672428006847'), ('title', u'Fishing for errors in the\xa0tests'), ('impactStatement', u'Testing a document which mimics the format of a file we
\u2019ve used \xa0before plus CO<sub>2</sub> and Ca<sup>2+</sup>.'), ('published', '2015-05-05T00:00:00Z'), ('image', OrderedDict([('thumbnail', OrderedDict([('uri', u'htt
ps://end2end--iiif.elifesciences.org/digests/9149295672428006847%2Fdigest-9149295672428006847.jpg'), ('alt', ''), ('source', OrderedDict([('mediaType', 'image/jpeg'), ('ur
i', u'https://end2end--iiif.elifesciences.org/digests/9149295672428006847%2Fdigest-9149295672428006847.jpg/full/full/0/default.jpg'), ('filename', u'digest-914929567242800
6847.jpg')])), ('size', OrderedDict([('width', 2000), ('height', 1181)]))]))])), ('subjects', [OrderedDict([('id', 'human-biology-medicine'), ('name', u'Human Biology and 
Medicine')])]), ('content', [OrderedDict([('type', 'image'), ('image', OrderedDict([('uri', u'https://end2end--iiif.elifesciences.org/digests/9149295672428006847%2Fdigest-
9149295672428006847.jpg'), ('alt', ''), ('source', OrderedDict([('mediaType', 'image/jpeg'), ('uri', u'https://end2end--iiif.elifesciences.org/digests/9149295672428006847%
2Fdigest-9149295672428006847.jpg/full/full/0/default.jpg'), ('filename', u'digest-9149295672428006847.jpg')])), ('size', OrderedDict([('width', 2000), ('height', 1181)]))]
)), ('caption', [OrderedDict([('type', 'paragraph'), ('text', u'<b>It\u2019s not just mammals who can recognise sample data.</b>\xa0Image credit:\xa0Anonymous and Anonymou
s\xa0(CC BY\xa04.0)')])])])]), ('relatedContent', [OrderedDict([('type', u'registered-report'), ('status', u'vor'), ('id', u'9149295672428006847'), ('version', 1), ('doi',
 u'10.7554/eLife.9149295672428006847'), ('authorLine', u'Irawati Kandela et al.'), ('title', u'Registered report: Discovery and preclinical validation of drug indications 
using compendia of public gene expression data'), ('stage', u'preview'), ('volume', 4), ('elocationId', u'e9149295672428006847')])]), ('stage', 'preview')])
2018-09-24T13:15:39Z INFO worker_4064 Response from digest API: 204
```

**Pull request merged**